pkg/systemd/generate/pods.go

```diff
@@ -266,6 +266,16 @@ func setPodExitPolicy(cmd []string) []string {
 	return append(cmd, "--exit-policy=stop")
 }
 
+// Make sure the name is specified by the --name argument
+func convertNameArg(cmd []string) []string {
+	for _, arg := range cmd {
+                if strings.HasPrefix(arg, "--name=") || arg == "--name" {
+                        return cmd
+                }
+        }
+	return append(cmd[:len(cmd)-1], "--name", cmd[len(cmd)-1])
```

Comment:
What if we have `podman pod create`?  I think we need to pass the real name of the pod to this function and do the transformation only when the last index matches that name.

Reply:
Right, I didn't think of that. Checking the name of the pod is a good idea, I'll try to familiarize myself with the codebase a bit more and see what I can do. Some extra test cases for this scenario wouldn't hurt either.